domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/FeatureFlagsLogger.kt

```diff
@@ -0,0 +1,129 @@
+package org.oppia.android.domain.oppialogger.analytics
+
+import org.oppia.android.app.model.EventLog
+import org.oppia.android.app.model.EventLog.FeatureFlagItemContext
+import org.oppia.android.app.model.EventLog.FeatureFlagListContext
+import org.oppia.android.util.platformparameter.APP_AND_OS_DEPRECATION
+import org.oppia.android.util.platformparameter.DOWNLOADS_SUPPORT
+import org.oppia.android.util.platformparameter.EDIT_ACCOUNTS_OPTIONS_UI
+import org.oppia.android.util.platformparameter.ENABLE_PERFORMANCE_METRICS_COLLECTION
+import org.oppia.android.util.platformparameter.EXTRA_TOPIC_TABS_UI
+import org.oppia.android.util.platformparameter.EnableAppAndOsDeprecation
+import org.oppia.android.util.platformparameter.EnableDownloadsSupport
+import org.oppia.android.util.platformparameter.EnableEditAccountsOptionsUi
+import org.oppia.android.util.platformparameter.EnableExtraTopicTabsUi
+import org.oppia.android.util.platformparameter.EnableFastLanguageSwitchingInLesson
+import org.oppia.android.util.platformparameter.EnableInteractionConfigChangeStateRetention
+import org.oppia.android.util.platformparameter.EnableLearnerStudyAnalytics
+import org.oppia.android.util.platformparameter.EnableLoggingLearnerStudyIds
+import org.oppia.android.util.platformparameter.EnablePerformanceMetricsCollection
+import org.oppia.android.util.platformparameter.EnableSpotlightUi
+import org.oppia.android.util.platformparameter.FAST_LANGUAGE_SWITCHING_IN_LESSON
+import org.oppia.android.util.platformparameter.INTERACTION_CONFIG_CHANGE_STATE_RETENTION
+import org.oppia.android.util.platformparameter.LEARNER_STUDY_ANALYTICS
+import org.oppia.android.util.platformparameter.LOGGING_LEARNER_STUDY_IDS
+import org.oppia.android.util.platformparameter.PlatformParameterValue
+import org.oppia.android.util.platformparameter.SPOTLIGHT_UI
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Convenience logger for feature flags.
+ *
+ * This logger is meant to be used for feature flag-related logging on every app launch. It is
+ * primarily used within the ApplicationLifecycleObserver to log the status of feature flags in a
+ * given app session.
+ */
+@Singleton
+class FeatureFlagsLogger @Inject constructor(
```

Comment:
I missed this in an earlier review, but how do we ensure that new features are added here? It seems critically important that at least a test fails if someone adds a feature flag and forgets to update this logger file.

Reply:
Two ways. We can do both the tests and add documentation to the wiki to guide the addition of future feature flags.

Reply:
I think tests would be the main way as documentation can easily be missed. Let's focus on that for this PR (continuing in that separate discussion thread).